netbsd/NetBSDProcess.c

```diff
@@ -0,0 +1,239 @@
+      .flags = 0,
+   },
+   [STARTTIME] = {
+      .name = "STARTTIME",
+      .title = "START ",
+      .description = "Time the process was started",
+      .flags = 0,
+   },
+   [PROCESSOR] = {
+      .name = "PROCESSOR",
+      .title = "CPU ",
+      .description = "Id of the CPU the process last executed on",
+      .flags = 0,
+   },
+   [M_VIRT] = {
+      .name = "M_VIRT",
+      .title = " VIRT ",
+      .description = "Total program size in virtual memory",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [M_RESIDENT] = {
+      .name = "M_RESIDENT",
+      .title = "  RES ",
+      .description = "Resident set size, size of the text and data sections, plus stack usage",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [ST_UID] = {
+      .name = "ST_UID",
+      .title = "  UID ",
+      .description = "User ID of the process owner",
+      .flags = 0,
+   },
+   [PERCENT_CPU] = {
+      .name = "PERCENT_CPU",
+      .title = "CPU% ",
+      .description = "Percentage of the CPU time the process used in the last sampling",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [PERCENT_NORM_CPU] = {
+      .name = "PERCENT_NORM_CPU",
+      .title = "NCPU%",
+      .description = "Normalized percentage of the CPU time the process used in the last sampling (normalized by cpu count)",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [PERCENT_MEM] = {
+      .name = "PERCENT_MEM",
+      .title = "MEM% ",
+      .description = "Percentage of the memory the process is using, based on resident memory size",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [USER] = {
+      .name = "USER",
+      .title = "USER      ",
+      .description = "Username of the process owner (or user ID if name cannot be determined)",
+      .flags = 0,
+   },
+   [TIME] = {
+      .name = "TIME",
+      .title = "  TIME+  ",
+      .description = "Total time the process has spent in user and system time",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [NLWP] = {
+      .name = "NLWP",
+      .title = "NLWP ",
+      .description = "Number of threads in the process",
+      .flags = 0,
+   },
+   [TGID] = {
+      .name = "TGID",
+      .title = "TGID",
+      .description = "Thread group ID (i.e. process ID)",
+      .flags = 0,
+      .pidColumn = true,
+   },
+};
+
+Process* NetBSDProcess_new(const Settings* settings) {
+   NetBSDProcess* this = xCalloc(sizeof(NetBSDProcess), 1);
+   Object_setClass(this, Class(NetBSDProcess));
+   Process_init(&this->super, settings);
+   return &this->super;
+}
+
+void Process_delete(Object* cast) {
+   NetBSDProcess* this = (NetBSDProcess*) cast;
+   Process_done((Process*)cast);
+   free(this);
+}
+
+static void NetBSDProcess_writeField(const Process* this, RichString* str, ProcessField field) {
+   char buffer[256]; buffer[255] = '\0';
+   int attr = CRT_colors[DEFAULT_COLOR];
+
+   switch (field) {
+   // add NetBSD-specific fields here
```

Comment:
Are there really only platform-independent fields?

Reply:
To the best of my knowledge yes, NetBSD does not have any platform-dependent fields like how FreeBSD has Jails and Jail ID (jid).